pkg/adapter/adapter.go

```diff
@@ -97,27 +103,38 @@ func (s *sinker) createRecorder(ctx context.Context, agentName string) record.Ev
 	return recorder
 }
 
-func (s *sinker) getHTTPClient(ctx context.Context) (*http.Client, error) {
+func (s *sinker) getHTTPClient() (*http.Client, error) {
 	var (
 		caCert    []byte
 		tlsConfig *tls.Config
 	)
-	clusterInterceptorList, err := clusterinterceptorsinformer.Get(s.injCtx).Lister().List(labels.NewSelector())
-	if err != nil {
-		return &http.Client{}, err
-	}
-	for i := range clusterInterceptorList {
-		if !bytes.Equal(clusterInterceptorList[i].Spec.ClientConfig.CaBundle, []byte{}) {
-			caCert = clusterInterceptorList[i].Spec.ClientConfig.CaBundle
-			certPool := x509.NewCertPool()
-			if ok := certPool.AppendCertsFromPEM(caCert); !ok {
-				return &http.Client{}, fmt.Errorf("unable to parse cert from %s", caCert)
-			}
-			tlsConfig = &tls.Config{
-				RootCAs:    certPool,
-				MinVersion: tls.VersionTLS13, // Added MinVersion to avoid  G402: TLS MinVersion too low. (gosec)
+	err := wait.PollImmediate(interval, timeout, func() (bool, error) {
+		clusterInterceptorList, err := clusterinterceptorsinformer.Get(s.injCtx).Lister().List(labels.NewSelector())
+		if err != nil {
+			return false, err
+		}
+		var count int
+		for i := range clusterInterceptorList {
+			if !bytes.Equal(clusterInterceptorList[i].Spec.ClientConfig.CaBundle, []byte{}) {
+				caCert = clusterInterceptorList[i].Spec.ClientConfig.CaBundle
+				certPool := x509.NewCertPool()
+				if ok := certPool.AppendCertsFromPEM(caCert); !ok {
+					return false, fmt.Errorf("unable to parse cert from %s", caCert)
+				}
+				tlsConfig = &tls.Config{
+					RootCAs:    certPool,
```

Comment:
Isn't this being overwritten everytime?

Reply:
Yes its overwritten

Need to change the logic of creation of http client

Reply:
Updated

Thank you :+1: